tests/layering/layering_test.go

```diff
@@ -0,0 +1,148 @@
+package e2e_test
+
+import (
+	"context"
+	"flag"
+	"testing"
+
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var deleteBuild = flag.Bool("delete-build", false, "Delete the derived build at the end of the test.")
+var deleteMachine = flag.Bool("delete-machine", false, "Delete the target machine after test run.")
+var streamBuild = flag.Bool("stream-build", false, "Stream the derived image build to stdout")
+var buildLogFile = flag.String("build-log", "", "Filename to write the derived image build log to")
+
+const (
+	imageStreamName = "test-boot-in-cluster-image"
+	buildName       = imageStreamName
+	// ostreeUnverifiedRegistry means no GPG or container signatures are used.
+	// Right now we're usually relying on digested pulls. See
+	// https://github.com/openshift/machine-config-operator/blob/master/docs/OSUpgrades.md#questions-and-answers around integrity.
+	// See https://docs.rs/ostree-ext/0.5.1/ostree_ext/container/struct.OstreeImageReference.html
+	ostreeUnverifiedRegistry = "ostree-unverified-registry"
+	imageRegistry            = "image-registry.openshift-image-registry.svc:5000"
+	// If this moves from /run, make sure files get cleaned up
+	authfilePath   = "/run/ostree/auth.json"
+	mcoNamespace   = "openshift-machine-config-operator"
+	imagePullSpec  = "registry.ci.openshift.org/rhcos-devel/rhel-coreos:%s"
+	helloWorldPath = "/usr/bin/hello-world"
+	imageURL       = ostreeUnverifiedRegistry + ":" + imageRegistry + "/" + mcoNamespace + "/" + imageStreamName
+)
+
+type Deployments struct {
+	Booted                  bool   `json:"booted"`
+	ContainerImageReference string `json:"container-image-reference"`
+}
+
+type Status struct {
+	Deployments []Deployments `json:"deployments"`
+}
+
```

Comment:
We really need to make this a library somewhere :flushed: 
https://github.com/jkyros/machine-config-operator/pull/1#discussion_r819106902

Reply:
We also have a duplicated `getRPMOStreeStatus`